New issue template

OmarShehata opened the pull request:
This is to resolve https://github.com/AnalyticalGraphicsInc/cesium/issues/7040.

I was hoping you could see it on my fork but it looks like you can't create new issues there. I realized you can comment stuff out with `<!--` so we don't need to say "please delete this message before posting".

I think this should help:

* Direct people to the forum with a nice interface.
* Remind people to post their browser/OS and a Sandcastle example if possible.

I didn't include "Cesium version" because I think it's usually implicit that it's the latest version/it doesn't seem like it's a problem that people are opening issues for older versions of Cesium. 

Feedback welcome!

cesium-concierge commented:
Thanks for the pull request @OmarShehata!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/issue-template/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

pjcozzi commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -0,0 +1,6 @@
+about: Please use the forum (https://groups.google.com/forum/#!forum/cesium-dev) for general questions about using Cesium. 
+---
+
+:exclamation: Please use the [forum](https://groups.google.com/forum/#!forum/cesium-dev) for asking questions about how to use Cesium and best practices. :exclamation:
```
Let's folks know that the core Cesium team monitors the forum and answers questions - to give them motivation and assurance to use it..

pjcozzi commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,10 @@
+<!-- 
+	Thanks for contributing to the Cesium Community! 
+
+	When suggesting an idea, give examples of the intended use case.
```
Maybe also add that contributions are welcome - and even better than a feature request - and link to the related part of the Contributor's Guide.

pjcozzi commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,12 @@
+Operating System:
+
+Browser:
+<!-- Creating a Sandcastle example (https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/) that reproduces the issue helps us a lot in tracking down bugs -->
```
The Sandcastle example / code example is the most important part.  The OS/browser matters, but less often so please put the code example request at the top.

OmarShehata commented:
Should be better now! I looked at a couple of other big projects, the one thing that stood out was I do like how [Tensorflow's template](https://github.com/tensorflow/tensorflow/issues/new) justifies why you shouldn't post support questions on there. Although I don't think it's a big enough problem here that it warrants making the wording more verbose.

pjcozzi commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,16 @@
+	Creating a Sandcastle example (https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/) that reproduces the issue helps us a lot in tracking down bugs. 
+--> 
+
+Sandcastle example reproducing the issue: 
```
Perhaps "Code example (use Share in Sandcastle to generate a url)"

pjcozzi commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,12 @@
+
+	When suggesting an idea, give examples of the intended use case.
+
+	If you're interested in seeing this feature in Cesium and want to work on it yourself,  we absolutely appreciate contributions! Check out the contributor's guide (https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md). We're happy to provide advice and feedback. 
```
For framing - why would someone want to contribute? They can get to it faster, fun, fame, etc.

pjcozzi commented on .github/ISSUE_TEMPLATE/question.md:
```diff
@@ -0,0 +1,6 @@
+---
+name: Ask a question
+about: Please use the forum (https://groups.google.com/forum/#!forum/cesium-dev) for general questions about using Cesium. 
```
Does the header accept markdown?

pjcozzi commented:
@OmarShehata this looks good, just a few tweaks to maybe make it a tad better.

OmarShehata commented:
@pjcozzi made some more tweaks. The header does not accept markdown unfortunately. 

For the Sandcastle example, I didn't want to put the instruction inline because I feel like it looks cluttered, so I put it in the comment right above. 

I'm open to any other edits. Feel free to tweak yourself or let me know.

pjcozzi commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,16 @@
+---
+
+<!--
+	Thanks for helping us improve Cesium! Please describe what the expected behavior is vs what actually happens. 
```
Last request: let's also encourage folks to contribute bug fixes.

OmarShehata commented:
@pjcozzi added a note to encourage users to contribute a bug fix. Let me know how the framing sounds.

I added it at the bottom because it's more important for the user to see the link to Sandcastle and how to share it. I didn't add it inside the same `<!-- -->` block because if you read "Sandcastle example:" first and scan back up to figure out the link to Sandcastle, it should be easy to find. 

I also removed the indents because I think it makes better use of the space in the editor box that way.
